mcu/src/mcu/uart_reader.py

```diff
@@ -0,0 +1,674 @@
+                    logger,
+                )
+                write_audio(
+                    sound_data,
+                    settings["sampling_frequency"],
+                    settings["audio_output_folder"],
+                    f"audio_{settings['sampling_frequency']}Hz",
+                    settings["audio_output_type"],
+                    settings["overwrite_audio"],
+                    logger,
+                )
+            else:
+                logger.info(f"{data}")
+
+
+@click.command()
+@click.option(
+    "-p", "--port", default="-- No COM --", help="The serial port to read data from."
+)
+@click.option(
+    "-b", "--baudrate", default=115200, help="The baudrate of the serial port."
+)
+@click.option(
+    "-s",
+    "--sampling-frequency",
+    default=10200,
+    help="The sampling frequency of the ADC.",
+)
+@click.option(
+    "-m", "--max-adc-value", default=4096, help="The maximum value of the ADC."
+)
+@click.option("-v", "--vdd", default=3.3, help="The voltage of the power supply.")
+@click.option(
+    "-o",
+    "--plot-output-type",
+    default="WEB",
+    type=click.Choice(["WEB", "FILE"]),
+    help="The type of output for the plot.",
+)
+@click.option(
+    "-l",
+    "--log-level",
+    default="INFO",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]),
+    help="The level of logging.",
+)
+@click.option(
+    "-f",
+    "--log-file",
+    is_flag=True,
+    help="Whether to log to a file (Not modifiable at runtime).",
+)
+@click.option(
+    "-w",
+    "--overwrite-audio",
+    is_flag=True,
+    help="Whether to overwrite the audio folder.",
+)
+@click.option(
+    "-a",
+    "--audio-output-type",
+    default="WAV",
+    type=click.Choice(["WAV", "OGG"]),
+    help="The type of output for the audio.",
+)
+@click.option(
+    "-d",
+    "--audio-output-folder",
+    default="audio_files",
+    help="The folder to save the audio files.",
+)
+@click.option("-c", "--cli", is_flag=True, help="Whether to run the CLI application.")
+def main(
+    port,
+    baudrate,
+    sampling_frequency,
+    max_adc_value,
+    vdd,
+    plot_output_type,
+    log_level,
+    log_file,
+    overwrite_audio,
+    audio_output_type,
+    audio_output_folder,
+    cli,
+):
+    settings = {
+        "port": port,
+        "baudrate": baudrate,
+        "sampling_frequency": sampling_frequency,
+        "max_adc_value": max_adc_value,
+        "vdd": vdd,
+        "plot_output_type": plot_output_type,
+        "log_level": log_level,
+        "log_file": log_file,
+        "overwrite_audio": overwrite_audio,
+        "audio_output_type": audio_output_type,
+        "audio_output_folder": audio_output_folder,
+    }
+
+    # Create the audio output folder if it doesn't exist
+    settings["audio_output_folder"] = os.path.join(
+        os.path.dirname(__file__), audio_output_folder
+    )
+    if not os.path.exists(audio_output_folder):
+        os.makedirs(audio_output_folder)
```

Comment:
Last comment: you should use `pathlib.Path` instead of `os.path.<function>`, see https://docs.python.org/3/library/pathlib.html#corresponding-tools.